pkg/apis/core/types.go

```diff
@@ -2160,6 +2160,25 @@ type ResourceRequirements struct {
 	// otherwise to an implementation-defined value
 	// +optional
 	Requests ResourceList
+	// Claims lists the names of resources, defined in spec.resourceClaims,
+	// that are used by this container.
+	//
+	// This is an alpha field and requires enabling the
+	// DynamicResourceAllocation feature gate.
+	//
+	// This field is immutable..
+	//
+	// +featureGate=DynamicResourceAllocation
+	Claims []ResourceClaim
+}
+
+// ResourceClaim references one entry in PodSpec.ResourceClaims.
+type ResourceClaim struct {
```

Comment:
I am not happy with the name of this struct anymore. After going back to "ResourceClaim" in resource.k8s.io, there are two structs with this name.

We can still rename it at any time (only affects Go clients, not the API), but perhaps we can already come up with a better name now? 

ResourceClaimRef? That sounds too much like an external reference.

SpecResourceClaim? That's better.

Reply:
It is a struct because because we might want to add additional per-entry parameters in the future, similar to mount parameters for volumes.

With "Claims []string" we cannot do that and would have to start using a new field name.

We could do "ClaimNames []string" now and switch to "Claims []ResourceClaimMount" in the future when the need arises, but then we would have to support both fields or switch before going beta. It seemed safer to use the slight more extensible API already now.

Let's not get hung up on the struct name. We can still switch to a nicer name later.

Reply:
I agree a different name would make it clearer.  `ResourceClaimMount` is "too cute".  `ResourceClaimAttachment` or `ResourceClaimUser` or `ResourceClaimUse` or similar is fine to me

Reply:
But I don't need to change this for alpha because its not visible in the API, right?

Reply:
that's fine.  It is visible in API docs, but I can live with it for the moment